### PR TITLE
Confusion matrix plotter & accurracy averager 

### PR DESCRIPTION
**Pull request by ZiRaymondal**
Hi,

Here is harry_plotter.py, a script which computes averages and display confusion matrix on 10-Fold X-Validation results
I've also put my results in the readme

Hope you'll find it useful (because averaging and parsing confusion matrix manually is boring)
Merry Xmas

### CODE REVIEW

File: recipes/UrbanSound8k/README.md [hunk 1]
```diff
@@ -54,7 +54,41 @@ python train.py hparams/train_ecapa_tdnn.yaml --data_folder=/localscratch/UrbanS
 python train.py hparams/train_ecapa_tdnn.yaml --data_folder=/localscratch/UrbanSound8K/ --train_fold_nums=[1, 2, 3, 4, 5, 6, 7, 8, 9] --valid_fold_nums=[10] --test_fold_nums=[10] --output_folder=./results/urban_sound/fold_10
 `
 
-Note that the results for 10-fold must be compiled from the output folders and averaged manually.
+Note that the results for 10-fold must be compiled from the output folders and averaged using ./SoundClassification/harry_plotter.py (which computes average and displays confusions matrix using seaborn) or manually.
+
+# Perfomance without data augmentation (10 folds)
+
+* Results averaged using [harry_plotter.py](https://github.com/ZiRaymondal/speechbrain/tree/develop/recipes/UrbanSound8k/SoundClassification/harry_plotter.py)
```
Comment (AsuMagic) [inline recipes/UrbanSound8k/README.md:61]: This should be a relative link within the same repo, not a link to your external repository.

File: recipes/UrbanSound8k/README.md [hunk 2]
```diff
@@ -54,7 +54,41 @@ python train.py hparams/train_ecapa_tdnn.yaml --data_folder=/localscratch/UrbanS
...
+
+|**Label\Precision** |**Speechbrain**|
```
Comment (AsuMagic) [inline recipes/UrbanSound8k/README.md:63]: Might be worth formatting the table for better markdown compliance, e.g. properly aligning the `|`s and adding a trailing `|` at the end.

File: recipes/UrbanSound8k/README.md [hunk 3]
```diff
@@ -54,7 +54,41 @@ python train.py hparams/train_ecapa_tdnn.yaml --data_folder=/localscratch/UrbanS
...
+|**street_music**    |  0.5960
+|**gun_shot**      |  0.7890           
+|**siren**         |  0.8759  
+|**engine_idling** |  0.6747              
+|**jackhammer**    |  0.5452
+|**drilling**      |  0.6761   
+|**car_horn**      |  0.7184
+|**Overall Mean Acc** |**0.72559**
+
+![Averaged confusion matrix](https://gitlab.com/Raymondaud.Q/lia-alternance-m2/-/raw/master/SpeechBrain-UrbanSound-Recipe/10_Fold/SANS_DATA_AUGMENTATION/Sum_Confu_SB_Default.png)
```
Comment (AsuMagic) [inline recipes/UrbanSound8k/README.md:77]: Same as above, this should be a relative link within the same repo, not a link to your external repository (so the image should be uploaded somewhere in your fork's branch, in whatever asset directory may be appropriate).

File: recipes/UrbanSound8k/README.md [hunk 4]
```diff
@@ -54,7 +54,41 @@ python train.py hparams/train_ecapa_tdnn.yaml --data_folder=/localscratch/UrbanS
...
+|**engine_idling** |  0.6747              
+|**jackhammer**    |  0.5452
+|**drilling**      |  0.6761   
+|**car_horn**      |  0.7184
+|**Overall Mean Acc** |**0.72559**
+
+![Averaged confusion matrix](https://gitlab.com/Raymondaud.Q/lia-alternance-m2/-/raw/master/SpeechBrain-UrbanSound-Recipe/10_Fold/SANS_DATA_AUGMENTATION/Sum_Confu_SB_Default.png)  
+
+* Results on each fold ( X = folds, Y = labels ) : 
+```
```
Comment (AsuMagic) [inline recipes/UrbanSound8k/README.md:80]: Could be presented as a table too, but this works.

**Comment by TParcollet** [comment]
Hi thanks for this PR. I wonder however if this couldn't be integrated in a more general way i.e. we try to avoid having recipe specific scripts. Do you think this work could be extended to work on multiple recipes?

**Comment by ZiRaymondal** [comment]
> Hi thanks for this PR. I wonder however if this couldn't be integrated in a more general way i.e. we try to avoid having recipe specific scripts. Do you think this work could be extended to work on multiple recipes?

Hmmm, it actually uses line number of train_log files on this specific recipe with this specific hparams file so, well no it couldn't be integrated more globally ( since plotting with seaborn confusion matrix should be done on a pc not on the gpu node etc :/ ).
Well, no problem it was a great fight with the linter :)

Merry xmas :D

**Comment by ZiRaymondal** [comment]
For information here is a repo ( in french ) where i've used harry_plotter.py in order to plot confusions and compute average score : https://gitlab.com/Raymondaud.Q/lia-alternance-m2/-/tree/master/SpeechBrain-UrbanSound-Recipe/10_Fold

**Comment by Adel-Moumen** [comment]
Hello, 

I'm closing this PR since no activity and modifications can be provided. 

Feel free to reopen if needed. 🙂

**Pull request closed without merging**